code/game/atoms.dm

```diff
@@ -156,6 +161,15 @@
 	if (canSmoothWith)
 		canSmoothWith = typelist("canSmoothWith", canSmoothWith)
 
+	if(custom_materials && custom_materials.len)
+		var/temp_list = list()
+		for(var/i in custom_materials)
+			var/datum/material/material = getmaterialref(i) || i
+			temp_list[material] = custom_materials[material] //Get the proper instanced version
+
+		custom_materials = null //Null the list to prepare for applying the materials properly
```

Comment:
So this sets the list to null before running set_custom_materials()

Reply:
Yes it runs it on the templ_list not the custom_materials